src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,103 @@
-import { useState } from 'react';
+/* eslint-disable max-len */
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+
+  function toTrim(string: string) {
+    return string.trim();
+  }
+
+  const [count, setCount] = useState(0);
+
+  const [newMovie, setNewMovie] = useState<Movie>({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+
+  const isDisabled = !toTrim(newMovie.title) || !toTrim(newMovie.imgUrl) || !toTrim(newMovie.imdbUrl) || !toTrim(newMovie.imdbId);
+
+  const reset = () => {
+    setNewMovie({
+      title: '',
+      description: '',
+      imgUrl: '',
+      imdbUrl: '',
+      imdbId: '',
+    });
+    setCount(count + 1);
+  };
+
+  const handleChange = (name: string, value: string) => {
+    setNewMovie({ ...newMovie, [name]: value });
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!newMovie.title || !newMovie.imgUrl || !newMovie.imdbUrl || !newMovie.imdbId) {
+      return;
+    }
+
+    onAdd(newMovie);
+    reset();
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={(event) => handleSubmit(event)}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={newMovie.title}
+        onChange={(newValue) => handleChange('title', newValue)}
```

Comment:
```suggestion
        onChange={handleChange}
```
no need to hardcode every field as name.
it will be event.target.name.
also see comments in TextField component (change type for onChange prop, pass name attribute and only refrerence to onChane in input)